### PR TITLE
Ipc connection

### PR DESCRIPTION
**Pull request by IceTank**

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ automatically logged in and validated against mojang's auth.
  * hideErrors : do not display errors, default to false
  * agent : a http agent that can be used to set proxy settings for yggdrasil authentication confirmation (see proxy-agent on npm)
  * validateChannelProtocol (optional) : whether or not to enable protocol validation for custom protocols using plugin channels for the connected clients. Defaults to true
+ * socketType (optional) : either `tcp` or `ipc`. Switches from a tcp connection to a ipc socket connection (or named pipes on windows). With the `ipc` option `host` becomes the path off the ipc connection on the local filesystem. Example: `\\.\pipe\minecraft-ipc` (Windows) `/tmp/minecraft-ipc.sock` (unix based systems). See the ipcConnection example for an example.
```
Comment (rom1504) [inline docs/API.md:34]: can't we just make port optional ?

Comment (IceTank) [reply docs/API.md:34]: Yeah sure

Comment (IceTank) [reply docs/API.md:34]: Well actually that won't work. `createServer` already has port as an optional argument. If port is not set it defaults to 25565. Making it do something else would be a breaking change.

**Comment by rom1504** [comment]
can you rebase

**Comment by rom1504** [comment]
please re-open if you want to finish it

**Pull request closed without merging**